File: dev/process-images.py
import os
import pathlib
from PIL import Image

MAX_WIDTH = 1750;

files = os.listdir(os.getcwd() + "/original-images")
for file in files:
    # skip annoying mac files
    if file == ".DS_Store":
        continue
    # file names
    base = file.split(".")[0]
    sourceFilePath = os.getcwd() + "/original-images/" + file
    targetImagePath = os.getcwd() + '/src/images/' + base + ".jpg"

    image = Image.open(sourceFilePath)
    imageWidth, imageHeight = image.size
    aspectRatio = imageWidth / imageHeight
    height = int(MAX_WIDTH / aspectRatio)
    image = image.resize((MAX_WIDTH, height))
    image = image.convert('RGB')
    image.save(targetImagePath)
    print("Successfully converted " + sourceFilePath + " to " + targetImagePath)

print("Done!")

File: dev/src/process-images.ts
import fs from 'fs';
import path from 'path';
import sharp from 'sharp';

const MAX_WIDTH = 1750;

const processImages = async () => {
  const originalImagesPath = path.join(__dirname, '../../original-images');
  const targetImagesPath = path.join(__dirname, '../../src/images');

  try {
    const files = await fs.promises.readdir(originalImagesPath);

    for (const file of files) {
      if (file === '.DS_Store') {
        continue;
      }

      const base = path.parse(file).name;
      const sourceFilePath = path.join(originalImagesPath, file);
      const targetImagePath = path.join(targetImagesPath, `${base}.jpg`);

      const image = sharp(sourceFilePath);
      const metadata = await image.metadata();
      
      const aspectRatio = metadata.width! / metadata.height!;

      if (aspectRatio === 0) {
        console.log(`\n\nWARNING! Skipping ${sourceFilePath} because it has no aspect ratio\n\n`);
        continue;
      }

      const height = Math.floor(MAX_WIDTH / aspectRatio);
      
      await image
        .resize(MAX_WIDTH, height)
        .toFile(targetImagePath);

      console.log(`Successfully converted ${sourceFilePath} to ${targetImagePath}`);
    }

    console.log('Done!');
  } catch (error) {
    console.error('Error processing images:', error);
  }
};

processImages();
